crates/bevy_pbr/src/meshlet/mod.rs

```diff
@@ -214,13 +207,28 @@ impl Plugin for MeshletPlugin {
                     Node3d::EndMainPass,
                 ),
             )
-            .init_resource::<MeshletMeshManager>()
             .insert_resource(InstanceManager::new())
-            .insert_resource(ResourceManager::new(
-                self.cluster_buffer_slots,
-                &render_device,
-            ))
-            .init_resource::<MeshletPipelines>()
+            .add_systems(
+                RenderStartup,
+                (
+                    check_meshlet_features,
+                    (
+                        (
+                            // This captures cluster_buffer_slots to create the resource manager.
+                            move |mut commands: Commands, render_device: Res<RenderDevice>| {
+                                commands.insert_resource(ResourceManager::new(
+                                    cluster_buffer_slots,
+                                    &render_device,
+                                ));
+                            },
+                            init_meshlet_pipelines,
+                        )
+                            .chain(),
+                        init_meshlet_mesh_manager,
+                    ),
+                )
+                    .chain(),
```

Comment:
this is a pretty dreadful amount of nesting, can any of it be split into a let binding or something?

Reply:
I realized I could split off the closure into a variable and I think this makes it much more tolerable. There's still the same amount of nesting, but I think it's easier to see what's happening.